core/types/block_ext_test.go

```diff
@@ -0,0 +1,105 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package types
+
+import (
+	"math/big"
+	"reflect"
+	"testing"
+	"unsafe"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCopyHeader(t *testing.T) {
+	t.Parallel()
+
+	t.Run("empty_header", func(t *testing.T) {
+		t.Parallel()
+
+		empty := &Header{}
+
+		headerExtra := &HeaderExtra{}
+		extras.Header.Set(empty, headerExtra)
+
+		cpy := CopyHeader(empty)
+
+		want := &Header{
+			Difficulty: new(big.Int),
```

Comment:
wow this comes to me as a surprise. I think we should fix this in OverrideNewEVMArgs so it sets difficulty to 0 not nil!

Reply:
https://github.com/ava-labs/coreth/pull/832